app/queues/certificate/dns-cleaner-worker.server.ts

```diff
@@ -9,8 +11,24 @@ export const dnsCleanerQueueName = 'certificate-cleanDns';
 export const dnsCleanerWorker = new Worker<CertificateJobData>(
   dnsCleanerQueueName,
   async (job) => {
-    const { rootDomain } = job.data;
-    logger.info(`TODO clean DNS for ${rootDomain}`);
+    const { rootDomain, certificateId } = job.data;
+
+    logger.info('Removing challenge DNS records', { rootDomain, certificateId });
+
+    await challengeModel.deleteChallengesByCertificateId(certificateId);
+
+    /**
+     * No need to manually delete the dns records as
+     * the database relation is set to cascade - challenge delete
+     * will remove the row from the dnsRecords table as well
+     *
+     * After those records were cascade deleted, we can trigger
+     * the reconciler as well
+     */
+
+    await setIsReconciliationNeeded(true);
+
+    logger.info('Challenges removed (cascading to records), reconciler triggered');
```

Comment:
```suggestion
    logger.info('Challenges removed (cascading to dns records), reconciler triggered');
```
Slightly more consistent with rest of the code